File: client/src/pages/RiderPage/models/RiderOverview.ts
import { z } from "zod";

export const riderRaceOverviewRowSchema = z.object({
    raceId: z.number(),
    raceName: z.string(),
    year: z.number(),
    active: z.boolean(),
    dnf: z.boolean(),
    price: z.number(),
    tooExpensive: z.boolean(),
    totalPoints: z.number(),
    totalParticipants: z.number(),
    selectedCount: z.number(),
    gcPosition: z.number().nullable(),
    pointsPosition: z.number().nullable(),
    komPosition: z.number().nullable(),
    youthPosition: z.number().nullable(),
});
export type RiderRaceOverviewRow = z.infer<typeof riderRaceOverviewRowSchema>;

export const riderSelectionHistoryRowSchema = z.object({
    username: z.string(),
    selectedCount: z.number(),
    totalCount: z.number(),
    races: z.array(z.string()),
});
export type RiderSelectionHistoryRow = z.infer<typeof riderSelectionHistoryRowSchema>;

export const riderOverviewSchema = z.object({
    riderId: z.number(),
    firstname: z.string(),
    lastname: z.string(),
    races: z.array(riderRaceOverviewRowSchema),
    selectionHistory: z.array(riderSelectionHistoryRowSchema),
});
export type RiderOverview = z.infer<typeof riderOverviewSchema>;

File: api/Models/Response/RiderOverview.cs
namespace SpoRE.Models.Response;

public record RiderRaceOverviewRow(
    int RaceId,
    string RaceName,
    int Year,
    bool Active,
    bool Dnf,
    int Price,
    bool TooExpensive,
    int TotalPoints,
    int TotalParticipants,
    int SelectedCount,
    int? GcPosition,
    int? PointsPosition,
    int? KomPosition,
    int? YouthPosition);

public record RiderSelectionHistoryRow(
    string Username,
    int SelectedCount,
    int TotalCount,
    IEnumerable<string> Races);

public record RiderOverview(
    int RiderId,
    string Firstname,
    string Lastname,
    IEnumerable<RiderRaceOverviewRow> Races,
    IEnumerable<RiderSelectionHistoryRow> SelectionHistory);

